### PR TITLE
feat(auto_assign): Add GitHub Actions workflow for auto-assigning PR creators

### PR DESCRIPTION
**Pull request by developerjhp**
## Changes

This PR introduces a GitHub Actions workflow (`auto_assign.yaml`) to automatically assign the PR creator as the assignee when a pull request is opened. It uses the `actions/github-script` to interact with GitHub's REST API.

## Visuals

No visuals are included, as this is a backend configuration update.

## Checklist
- [ ] Have you written the functional specifications?
- [ ] Have you written the test code?

## Additional Discussion Points

- Are there any additional edge cases we should test for? 
- Error handling and permission optimization could be reviewed for further improvement.

## Summary by CodeRabbit

- **New Features**
	- 풀 리퀘스트 생성 시 자동으로 작성자를 담당자로 지정하는 GitHub Actions 워크플로우를 추가했습니다.
	- `auto_assign.js` 파일에서 PR 작성자를 자동으로 할당하는 기능을 구현했습니다.
- **Chores**
	- `auto-assign` 프로젝트를 위한 `package.json` 파일을 추가했습니다.
	- VSCode 추천 확장 목록에 `"github.vscode-github-actions"`를 추가했습니다.

### CODE REVIEW

File: .github/workflows/auto_assign.yaml
Comment (noahluftyang) [inline .github/workflows/auto_assign.yaml:1]: 액션의 유지보수성을 높이기 위해 [JavaScript action](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-javascript-action?learn=create_actions&learnProduct=actions)으로 만들어서 관리하면 어떨까? ㅎㅎ

Comment (developerjhp) [reply .github/workflows/auto_assign.yaml:1]: 오 좋은생각인 것 같아!!
한번 시도해볼게!!

File: .github/workflows/auto_assign.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+      with:
```
Comment (froggy1014) [inline .github/workflows/auto_assign.yaml:24]: 이거 indent가 잘못들어간거 같은데? 

가끔 이런 syntax에러는 나같은 경우는

- https://github.com/redhat-developer/vscode-yaml/issues
- https://marketplace.cursorapi.com/items?itemName=GitHub.vscode-github-actions

익스텐션 설치해서 사용하는데, 알아보긴하는데 .vscode extensions.json에 추가하는게 좋을까 ?

File: .github/workflows/auto-assign/auto_assign.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-undef */
+/* eslint-disable @typescript-eslint/no-require-imports */
+const github = require('@actions/github');
+
+async function getOctokitAndContext() {
```
Comment (froggy1014) [inline .github/workflows/auto-assign/auto_assign.js:5]: 이런식으로 만들어서 쓰는거 처음알아따..

File: .github/workflows/auto_assign.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        cache: pnpm
+        node-version-file: .nvmrc
+
+      - name: Install dependencies for auto-assign
+        run: |
+          cd .github/workflows/auto-assign
+          npm ci
```
Comment (froggy1014) [inline .github/workflows/auto_assign.yaml:31]: pnpm ci 로 해야하는거 아닌가 ~ ? 

`pnpm install --frozen-lockfile`

File: .github/workflows/auto_assign.yaml (hunk 1 above)
Comment (Huinno-ParkJinHyun) [reply .github/workflows/auto_assign.yaml]: extension recommendations에 github-action 추가했어!! 고마워~!!

File: .github/workflows/auto_assign.yaml (hunk 3 above)
Comment (Huinno-ParkJinHyun) [reply .github/workflows/auto_assign.yaml]: 요것도 추가했어!! 고마웡~!!!

File: .github/workflows/auto_assign.yaml (hunk 1 above)
Comment (developerjhp) [reply .github/workflows/auto_assign.yaml]: extension recommendations에 github-action 추가했어!! 고마워~!!

File: .github/workflows/auto_assign.yaml (hunk 3 above)
Comment (developerjhp) [reply .github/workflows/auto_assign.yaml]: 요것도 추가했어!! 고마웡~!!!

**Review by froggy1014 — APPROVED** [review]
LGTM

**Review by hy57in — APPROVED** [review]
